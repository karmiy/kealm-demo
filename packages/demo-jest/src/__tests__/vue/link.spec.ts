import Vue from 'vue';
import { mount } from '@vue/test-utils';
import Link from '@components/vue/link';

describe('Link', () => {
    it('test snapshot that Link changes the class when hovered', async () => {
        const wrapper = mount(Link, {
            propsData: {
                page: 'https://github.com/karmiy/demo-jest',
            }
        });
        expect(wrapper.element).toMatchSnapshot();

        wrapper.trigger('mouseenter'); // 触发 mouseenter 事件
        await Vue.nextTick(); // 等待 nextTick 渲染
        expect(wrapper.element).toMatchSnapshot();

        wrapper.trigger('mouseleave'); // 触发 mouseenter 事件
        await Vue.nextTick(); // 等待 nextTick 渲染
        expect(wrapper.element).toMatchSnapshot();
    });
});
