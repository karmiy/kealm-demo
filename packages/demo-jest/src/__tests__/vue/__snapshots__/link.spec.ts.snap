// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link test snapshot that Link changes the class when hovered 1`] = `
<a
  class="normal"
  href="https://github.com/karmiy/demo-jest"
/>
`;

exports[`Link test snapshot that Link changes the class when hovered 2`] = `
<a
  class="hovered"
  href="https://github.com/karmiy/demo-jest"
/>
`;

exports[`Link test snapshot that Link changes the class when hovered 3`] = `
<a
  class="normal"
  href="https://github.com/karmiy/demo-jest"
/>
`;
